src/core/TESTING.md

```diff
@@ -62,8 +62,257 @@ test('my test', async () => {
 ### Strategies for specific Core APIs
 
 #### HTTP Routes
+The HTTP API interface is another public contract of Kibana. Although not every Kibana endpoint is for external use. Evaluating the required level of test coverage for an HTTP resource, make your judgment based on the fact whether an endpoint is considered to be public or private. Public API is expected to have a higher level of test coverage.
+Public API tests should cover **observable behavior** of the system. Therefore they should be close to the real user interactions as much as possible. Ideally, to use HTTP requests to communicate with the Kibana server.
+
+##### Integration tests
+Depending on the number of external dependencies, you can consider implementing several high-level integration tests. They would work as a set of [smoke tests](https://en.wikipedia.org/wiki/Smoke_testing_(software)) for the most important functionality.
+Subjects for tests:
+- authenticated / unauthenticated access to an endpoint.
+- endpoint validation (params, query, body).
+- main business logic.
+
+###### Functional Test Runner
+If your plugin leverages elasticsearch server to store data and supports additional configuration, you can utilize Functional Test Runner(FTR) to add integration tests. 
```

Comment:
> If your plugin leverages elasticsearch server to store data and supports additional configuration, you can utilize Functional Test Runner(FTR) to add integration tests. 

If your plugin relies on the elasticsearch server to store data and supports additional configuration, you can leverage the  Functional Test Runner(FTR) to implement integration tests.